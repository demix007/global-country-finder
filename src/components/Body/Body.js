import React from 'react';
import Countries from '../Countries/Countries';
import Filter from '../Filter/Filter';

const Body = () => (
  <div>
    <Filter />
    <Countries />
  </div>
);

export default Body;
