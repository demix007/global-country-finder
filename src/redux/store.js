import { configureStore } from '@reduxjs/toolkit';
import CountriesReducer from './reducer';

const store = configureStore({
  reducer: {
    countries: CountriesReducer,
  },
});

export default store;
