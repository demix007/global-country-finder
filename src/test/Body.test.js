import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/store';
import Body from '../components/Body/Body';

describe('Body', () => {
  it('Body component is rendered correctly', () => {
    const body = render(
      <Provider store={store}>
        <Body />
      </Provider>,
    );
    expect(body).toMatchSnapshot();
  });
});
