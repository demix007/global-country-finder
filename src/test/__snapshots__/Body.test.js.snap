// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Body Body component is rendered correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="search-container"
        >
          <form
            class="form__input"
          >
            <input
              class="search-input"
              id="search"
              name="search"
              placeholder="Search for a country"
              type="search"
              value=""
            />
            <button
              aria-label="search-country"
              class="search-btn"
              type="submit"
            >
              <i
                class="fa-sharp fa-solid fa-magnifying-glass-location"
              />
            </button>
          </form>
        </div>
        <div
          class="grid"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="search-container"
      >
        <form
          class="form__input"
        >
          <input
            class="search-input"
            id="search"
            name="search"
            placeholder="Search for a country"
            type="search"
            value=""
          />
          <button
            aria-label="search-country"
            class="search-btn"
            type="submit"
          >
            <i
              class="fa-sharp fa-solid fa-magnifying-glass-location"
            />
          </button>
        </form>
      </div>
      <div
        class="grid"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
